--- a/build.zig
+++ b/build.zig
@@ -775,7 +775,7 @@
         // This number should never be raised. If the value is exceeded, then
         // it is considered a bug in the Zig project that building the
         // compiler takes more than 8G of memory.
-        .max_rss = 8_000_000_000,
+        .max_rss = 7_500_000_000,
         .root_module = addCompilerMod(b, options),
     });
     exe.stack_size = stack_size;
@@ -833,7 +833,11 @@
     assert(cfg.lld_include_dir.len != 0);
     mod.addIncludePath(.{ .cwd_relative = cfg.lld_include_dir });
     mod.addIncludePath(.{ .cwd_relative = cfg.llvm_include_dir });
-    mod.addLibraryPath(.{ .cwd_relative = cfg.llvm_lib_dir });
+    // Split system_lib_dir by semicolon and add each path
+    var it = std.mem.splitScalar(u8, cfg.llvm_lib_dir, ';');
+    while (it.next()) |path| {
+        mod.addLibraryPath(.{ .cwd_relative = path });
+    }
     addCMakeLibraryList(mod, cfg.clang_libraries);
     addCMakeLibraryList(mod, cfg.lld_libraries);
     addCMakeLibraryList(mod, cfg.llvm_libraries);
